sdk/file_sdk/XMPFileFormat/xmpio_over_istream.cpp

```diff
@@ -0,0 +1,85 @@
+#include "xmpio_over_istream.h"
+
+namespace mip {
+namespace file {
+
+XMPIOOverIStream::XMPIOOverIStream(std::shared_ptr<IStream> stream) {
+  mBaseStream = stream;
+}
+
+XMPIOOverIStream::~XMPIOOverIStream() {
+  if (mTemp != nullptr)
+    delete mTemp;

```

Comment:
no need to check for null before delete. it is safe to delete null pointer

Reply:
i didn't know that. its not what we learn in school :(

---
In reply to: [137461676](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r137461676) [](ancestors = 137461676)

Reply:
Sorry, i had a merge problem and did all my changes again and missed some files

---
In reply to: [137476523](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r137476523) [](ancestors = 137476523,137461676)